sdk/src/Core/Internal/Emitters/UdpSegmentEmitter.cs

```diff
@@ -47,7 +48,7 @@ public UdpSegmentEmitter() : this(new JsonSegmentMarshaller())
         private UdpSegmentEmitter(ISegmentMarshaller marshaller)
         {
             _marshaller = marshaller;
-            _udpClient = new UdpClient();
+            _udpClient = new UdpClient();
             DaemonConfig daemonEndPoint = DaemonConfig.GetEndPoint();
             EndPoint = daemonEndPoint.UDPEndpoint;
```

Comment:
In the current implementation, `EndPoint = daemonEndPoint.UDPEndpoint;`, whenever, hostname is refreshed, the udp emitter `Endpoint` is not refreshed since `get` on `daemonEndpoint` is called only once, here.

Instead, we need to store `DaemonConfig` instance as a private member of this class. 
`private DaemonConfig _daemonConfig`.
In constructor, it would be : `_daemonConfig = DaemonConfig` and same [here](https://github.com/aws/aws-xray-sdk-dotnet/blob/36c1c83e3d8fdf49763ffe15be84facd3d4cf45c/sdk/src/Core/Internal/Emitters/UdpSegmentEmitter.cs#L147).

And this [point](https://github.com/aws/aws-xray-sdk-dotnet/blob/36c1c83e3d8fdf49763ffe15be84facd3d4cf45c/sdk/src/Core/Internal/Emitters/UdpSegmentEmitter.cs#L71), the code change is ` _udpClient.Send(data, data.Length, _daemonConfig.UDPEndpoint);`.

At `Debug` level we can have this log `_logger.DebugFormat("UDP Segment emitter enpoint: " + _daemonConfig.UDPEndpoint);`
This ensures, UDP emitter is updated in correct manner.

Reply:
Sorted in d95c7b0c698aac7808595a2c200fc1dc78561af4.